### ISSUE TITLE
Add a small section about iterative process in getting started page

### ISSUE THREAD

**Issue by kytrinyx** [exercism/exercism.io#2090]
From my email inbox:

> Incidentally, one thing that wasn't clear when I did the first exercise was
> the idea of the iterations. Basically until I submitted the first one, I
> didn't realize the system was setup to allow for multiple submissions per
> exercise. [...] Not a big
> deal, but something to consider is maybe adding language to the '
> http://exercism.io/getting-started' page that talks explicitly about making
> multiple submissions to try different solutions.

**Comment by outkaj** [comment]
I don't think you would necessarily need to add a separate paragraph - I also thought the idea of multiple iterations/coexisting versions was very clear on the website from the Ruby leap example. Perhaps just an additional line or two for clarification.

**Comment by kytrinyx** [comment]
Yeah, that sounds good.

**Comment by tejasbubane** [comment]
Since we don't have a [getting-started page](http://exercism.io/getting-started) now, I think we should add a sentence or two, just mentioning `iterations` on [the landing page](http://exercism.io/) and a detailed section on the [getting started for newbies page](http://exercism.io/how-it-works/newbie).

**Comment by kytrinyx** [comment]
There are two getting started pages:

- http://exercism.io/how-it-works/newbie
- http://exercism.io/how-it-works

The newbie page says:

> If you get ideas for how to improve your solution, then change your code, and submit it again.

Do you have a suggestion for how to rephrase this to clarify the idea of iterations?

**Comment by tejasbubane** [comment]
* http://exercism.io/how-it-works/ this page says:

> 1. Iterate on your solution. Integrate feedback. Try other approaches.
> 2. Browse other solutions to the same problem. Provide feedback: ask questions, suggest 
>     improvements, and ponder style and readability.

This, I think is pretty good for experienced developers. 

* http://exercism.io/how-it-works/

The last line on this page 

> Once you've gotten all of the tests passing, you're good to go. Submit the solution to the site using the CLI.

gives an impression like you submit the exercise and you are done.
 
I was thinking of more like a section at the end (after `Working the Exercises` section). This will add more info for new people and also complete the flow like

```
1. create-account 
2. install CLI
3. configure CLI 
4. choose language 
5. work on exercises 
6. iterate
```

**Comment by kytrinyx** [comment]
Yepp, you're right. It does sound like you're kind of done at that point. We should definitely rework it.

I like the suggested section. I might even expand 5 and 6 so that it's clear that you iterate on individual exercises.

**Comment by siakaramalegos** [comment]
It looks like the http://exercism.io/how-it-works page has been made more clear, but the newbie one hasn't yet. I can take a stab at it if you want. Should I post suggestions here first before actually doing it in code?

**Comment by siakaramalegos** [comment]
When I see that page (http://exercism.io/how-it-works/newbie), I feel like it's a large block of text. I bet most people don't read it all so the message about iteration is probably getting lost. 

There are a few design strategies that may help:
- Mini-navigation menu on the left to quickly jump to different sections or at least understand what all is discussed on this page
- Tabbed area for the actual steps, after the description of how it works
- Using a muted color for the notes on the right (like Bootstrap text-muted class)
- Splitting the page into 2 more major headings - Overview/How It Works and Getting Started
- General language clean-up to make it more clear and potentially reduce the amount of text

I'll work on just making the iteration aspect more clear in both the overview (top) section and in the working the exercises (bottom) section for now, but let me know if any of those thoughts or ideas make sense to you.

**Comment by siakaramalegos** [comment]
I may have found a potential bug in the set up - if this doesn't look familiar, I can submit another issue. I think 2 things may have happened. The bin/setup file doesn't prepend  `rake db:from_scratch` with `bundle exec` which my system didn't like for this repo. But now, I'm also getting an undefined method success? for nil:nilClass from 2 different lines in db.rake - basically `$CHILD_STATUS` seems to be nil.

I manually updated bin/setup on GitHub and re-cloned so it's not complaining about the rake command anymore but I am still getting the nil:nilClass errors for $CHILD_STATUS.

**Comment by kytrinyx** [comment]
> When I see that page (http://exercism.io/how-it-works/newbie), I feel like it's a large block of text. I bet most people don't read it all so the message about iteration is probably getting lost.

Yes! I just saw a talk called "design for non-designers" and I realized that I've definitely got far to many big blocks of text. Totally not scannable.

The two suggestions that I think we could probably just go for without any experimentation are

- using a muted color on the right, and
- general language cleanup.

The other options are various ways of recombining the content to be clearer, which might take some experimentation. How about playing around with various combinations locally and then posting some screenshots here to discuss which direction(s) seem clearer?

**Comment by siakaramalegos** [comment]
I did post a screenshot on the pull request with that exact first pass (muted color, language cleanup). I'm pasting it in below for convenience. But maybe instead of an iterative change, we should blow the whole thing up (other instruction pages too), and ask from a design standpoint, how should it look? Then, make them all consistent. Modest use of images (or even video) may help.

I don't even like front end but I'm a process engineer at heart so when people have a hard time figuring out what the process is, I always want to jump in to improve it. 

![exercism io](https://cloud.githubusercontent.com/assets/4777393/15455322/313687f2-2016-11e6-8b73-5b943e91babf.png)

**Comment by kytrinyx** [comment]
> I posted a screenshot on the pull request with that exact first pass (muted color, language cleanup).

:heart: great minds. I do recall seeing that now that you mention it.

I would love to start from scratch and talk about process (it's not my strong point, as you may have noticed). I feel like we get a first pass, and then we can discuss a better approach in the discussions repository (videos, blowing up the whole process, etc). If we could use this to combine experienced/newbie without frustrating either group that would be stellar.

**Comment by nickborromeo** [comment]
Do you all think that this issue can be closed? 😄 

I see the text `Improve your solution and resubmit as many times as desired.` on the newbie page and I think that is what this issue was initially opened for. It makes it a little clearer that there is an iterative process.

**Comment by kytrinyx** [comment]
Yeah, I think that's what it was meant to address. If not, I'm sure someone will open a new issue.
